rs/ethereum/cketh/minter/src/eth_rpc_client/mod.rs

```diff
@@ -276,15 +276,24 @@ impl EthRpcClient {
     pub async fn eth_send_raw_transaction(
         &self,
         raw_signed_transaction_hex: String,
-    ) -> HttpOutcallResult<JsonRpcResult<SendRawTransactionResult>> {
+    ) -> Result<SendRawTransactionResult, MultiCallError<SendRawTransactionResult>> {
+        if let Some(evm_rpc_client) = &self.evm_rpc_client {
+            return evm_rpc_client
+                .eth_send_raw_transaction(raw_signed_transaction_hex)
```

Comment:
As indicated in the PR description, there is a change of behaviour (also visible in the modification of the integration tests in `flow.rs`), since with the EVM-RPC canister this call will query all 3 providers in parallel. We could build the same `sequential_call_until_ok` logic on top of the EVM-RPC canister, but I have the impression that this is currently not worth it:

1. The minter doesn't send that many transactions (ballpark 1 tx per day)
2. Such logic would be best built in the EVM-RPC canister directly.

WDYT?

Reply:
> Such logic would be best built in the EVM-RPC canister directly.

Agree!

Reply:
I'm not sure if this "optimization" should ever be introduced. If you send the transaction to one provider and the provider tells you that it sent the transaction, what guarantee do you have that this is true?

Unless it is too costly, I'd prefer to always send it in parallel to all providers.

Reply:
> If you send the transaction to one provider and the provider tells you that it sent the transaction, what guarantee do you have that this is true?

Absolutely none. The minter will keep resubmitting (because the latest transaction count will not have increased) and of course the same provider could still keep lying. The fix would be easy, in that we could pseudo-randomize the order of the 3 providers, but that was never a problem so far (and hence not done).

Initially, when we built the ckETH minter, we were unsure what kind of load we were going to have for withdrawals,
and so we try to minimise the minter cycles consumptions by avoiding unnecessary HTTPs outcalls.

With our current experience, I think it makes sense to keep it simple and just use the out-of-the-box parallel strategy of the EVM-RPC canister.

Reply:
Closing this thread to merge. Feel free to ping me if there is anything left to discuss.